src/oom/oomd.c

```diff
@@ -142,10 +142,8 @@ static int run(int argc, char *argv[]) {
                 return log_error_errno(r, "Failed to get SwapTotal from /proc/meminfo: %m");
 
         r = safe_atollu(swap, &s);
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse SwapTotal from /proc/meminfo: %s: %m", swap);
-        if (s == 0)
-                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Requires swap to operate");
+        if (r < 0 || s == 0)
+                log_warning("Swap is currently not detected; memory pressure usage will be degraded");
```

Comment:
Perhaps log_info is enough? It's most likely a conscious configuration decision after all, it's unlikely that swap disappears unintentionally

Reply:
I think a log_warning is reasoanble. The only case where swap wouldn't be useful would be with a workload that either is overprovisioned or tailored to available memory. But in those cases, oomd is also not needed. In all other cases, at least zram should be used.